website/content/docs/agent/autoauth/methods/jwt.mdx

```diff
@@ -17,3 +17,7 @@ JWT to perform a reauthentication.
 - `path` `(string: required)` - The path to the JWT file
 
 - `role` `(string: required)` - The role to authenticate against on Vault
+
+- `remove_jwt_after_reading` `(bool: optional, defaults to true)` -
```

Comment:
Can we flip this around and have it be `keep_after_reading` or something along those lines? This way the default value is also the zero value (`false`).

Reply:
I thought about that but I thought it more prudent to retain the current default behavior and opt-out if needed. I'd assume that in _most_ cases, the default behavior is more appropriate. This change is mostly to allow for using Kubernetes bound service account tokens with JWT auth. In this use case, Kubernetes automatically rotates the JWT based on a configurable TTL.

Reply:
`keep_after_reading` with it being defaulted `false` should still retain the current behavior though. Note that we're flipping the naming and also the default so keeping the files would still be an opt-in. We'd only _not_ delete the file if this gets set to `true` explicitly.

Reply:
Ah ok, I see what you mean. I was following the same model that the AppRole auto-auth was using but I can change this around if it makes more sense.

Reply:
I actually like `remove_jwt_after_reading` for its consistency with AppRole's option: https://www.vaultproject.io/docs/agent/autoauth/methods/approle#remove_secret_id_file_after_reading

IMO the bar to break from this pattern needs to be higher than making it terser. If it wasn't for the prior art, I would 100% agree with both the flip + naming comments above, but they're minor nits and they're secondary to preventing confusion for consumers.